### PR TITLE
Expand docs badges slot

### PR DESCRIPTION
**Pull request by benjgil**
## Proposed Changes

- add ability to add badges to all sections of component-meta section (header) in docs

### CODE REVIEW

**Comment by itaymendel** [comment]
I feel like a better approach would be to have different slots for each, and not use a prop. @KutnerUri ?

File: scopes/component/ui/component-meta/component-overview.tsx [hunk 1]
```diff
@@ -44,30 +53,59 @@ export function ComponentOverview({
     const origin = isBrowser ? window.location.origin : undefined;
     finalElementsUrl = origin && elementsUrl ? `${origin}${elementsUrl}` : undefined;
   }
+
+  type CompWithTitleBadgesProps = { position: TitleBadgePosition; children: ReactNode };
+  function ComponentWithTitleBadges({ position, children }: CompWithTitleBadgesProps) {
```
Comment (KutnerUri) [inline scopes/component/ui/component-meta/component-overview.tsx:58]: extract outside of ComponentOverview, pass props (like titleBadges) explicitly

File: scopes/component/ui/component-meta/component-overview.tsx [hunk 2]
```diff
@@ -44,30 +53,59 @@ export function ComponentOverview({
...
+    return (
+      <div className={styles.componentWithBadgeContainer}>
+        {children}
```
Comment (KutnerUri) [inline scopes/component/ui/component-meta/component-overview.tsx:61]: do we need the container? I think you can just make a Badges component and then use it like so:

```
  <div className={styles.componentTitle}>
    <H1>{displayName}</H1>
      <Badges value={titleBadges} position={TitleBadgePosition.Title}
  </div>
```

File: scopes/component/ui/component-meta/component-overview.tsx [hunk 3]
```diff
@@ -44,30 +53,59 @@ export function ComponentOverview({
...
+            ?.sort((a, b) => a?.weight - b?.weight)
+            ?.map((titleBadge, index) => {
+              return (
+                <titleBadge.component
+                  key={index}
+                  componentDescriptor={componentDescriptor}
+                  legacyComponentModel={component}
+                />
+              );
+            })}
```
Comment (KutnerUri) [inline scopes/component/ui/component-meta/component-overview.tsx:81]: memo this (I know it already exists in the original code, but it needs memo-ing)

File: scopes/component/ui/component-meta/component-overview.tsx [hunk 4]
```diff
@@ -11,9 +11,18 @@ import { isBrowser } from '@teambit/ui-foundation.ui.is-browser';
 import { ComponentModel } from '@teambit/component';
 import styles from './component-overview.module.scss';
 
+export enum TitleBadgePosition {
```
Comment (KutnerUri) [inline scopes/component/ui/component-meta/component-overview.tsx:14]: rename - BadgePosition

**Review by KutnerUri — CHANGES_REQUESTED** [review]
cool PR. please add screenshots

File: scopes/component/ui/component-meta/component-overview.module.scss [hunk 5]
```diff
@@ -1,12 +1,16 @@
-.componentTitle {
+.componentWithBadgeContainer {
   display: flex;
   align-items: center;
   flex-wrap: wrap;
   margin-bottom: 20px;
```
[the remark this answers is not in the document]
Comment (benjgil) [reply scopes/component/ui/component-meta/component-overview.module.scss]: done

File: scopes/docs/docs/docs.ui.runtime.tsx [hunk 6]
```diff
@@ -15,8 +15,8 @@ export class DocsUI {
   /**
    * register a new title badge into the overview section of a component.
    */
-  registerTitleBadge(...titleBadge: TitleBadge[]) {
-    this.titleBadgeSlot.register(titleBadge);
+  registerTitleBadge(titleBadges: TitleBadge[]) {
```
Comment (GiladShoham) [inline scopes/docs/docs/docs.ui.runtime.tsx:18]: this change might be a breaking change for anyone that only pass one badge not as array

**Comment by benjgil** [comment]
image

File: scopes/docs/docs/docs.ui.runtime.tsx (hunk 6 above)
Comment (benjgil) [reply scopes/docs/docs/docs.ui.runtime.tsx]: reverting
Out of interest, is there any advantage in doing it like this? Just that you can enter a single item without making it an array?

File: scopes/component/ui/component-meta/component-overview.tsx (hunk 1 above)
Comment (benjgil) [reply scopes/component/ui/component-meta/component-overview.tsx]: done

File: scopes/component/ui/component-meta/component-overview.tsx (hunk 2 above)
Comment (benjgil) [reply scopes/component/ui/component-meta/component-overview.tsx]: removed

File: scopes/component/ui/component-meta/component-overview.module.scss [hunk 7]
```diff
@@ -1,12 +1,16 @@
-.componentTitle {
+.componentWithBadgeContainer {
```
[the remark this answers is not in the document]
Comment (benjgil) [reply scopes/component/ui/component-meta/component-overview.module.scss]: done (created a new Row component, in teambit.base-react

File: scopes/docs/docs/docs.ui.runtime.tsx (hunk 6 above)
Comment (benjgil) [reply scopes/docs/docs/docs.ui.runtime.tsx:18]: Unreverted, as without this change couldn't register multiple badges.
It's not clear why, as both with and without the change the register function is passed an array of badges... Unclear what the difference is, but it just doesnt work with the previous syntax.
Probably something to do with arrays and maps...

Comment (KutnerUri) [reply scopes/docs/docs/docs.ui.runtime.tsx:18]: @benjgil - you need to use it like this:
```ts
docs.registerTitleBadge(firstBadge, secondBadge, etcBadge);
```

and it has the advantage of being backward compatible and easier to read/write

File: scopes/docs/docs/docs.compare.section.tsx [hunk 8]
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
-import { TitleBadgeSlot } from '@teambit/docs';
 import { Section } from '@teambit/component';
 import { OverviewCompare } from '@teambit/docs.ui.overview-compare';
+import { DocsUI } from './docs.ui.runtime';
 
 export class OverviewCompareSection implements Section {
-  constructor(private titleBadgeSlot: TitleBadgeSlot) {}
+  constructor(private docs: DocsUI) {}
```
Comment (KutnerUri) [inline scopes/docs/docs/docs.compare.section.tsx:7]: why though? wouldn't it be simpler to pass just the slot? (also - as type)

File: scopes/docs/docs/docs.ui.runtime.tsx (hunk 6 above)
Comment (KutnerUri) [reply scopes/docs/docs/docs.ui.runtime.tsx:18]: please revert

Comment (KutnerUri) [reply scopes/docs/docs/docs.ui.runtime.tsx:18]: or allow single badges by checking `Array.isArray(badges)`

**Review by KutnerUri — CHANGES_REQUESTED** [review]
fix `registerBadge()` breaking change

File: scopes/docs/docs/docs.ui.runtime.tsx (hunk 6 above)
Comment (benjgil) [reply scopes/docs/docs/docs.ui.runtime.tsx:18]: we dont need backwards compatibility (i've changed the two places that currently used it) and the problem with your approach is that for whatever reason (myself and Gilad couldnt work it out with the short time we spent on it) it doesnt work when trying to add more than a single badge at a time.

Comment (benjgil) [reply scopes/docs/docs/docs.ui.runtime.tsx:18]: I dont see the utility of being able to add a single badge not as an array with one member, certainly not enough to refactor this again

**Review by KutnerUri — CHANGES_REQUESTED** [review]

File: scopes/component/ui/component-meta/component-overview.tsx [hunk 9]
```diff
@@ -44,32 +42,99 @@ export function ComponentOverview({
...
-                return (
-                  <titleBadge.component
-                    key={index}
-                    componentDescriptor={componentDescriptor}
-                    legacyComponentModel={component}
-                  />
-                );
-              })}
-          </div>
+        <div className={styles.titleRow}>
```
[the remark this answers is not in the document]
Comment (KutnerUri) [reply scopes/component/ui/component-meta/component-overview.tsx:49]: no, it's always bad to leave divs like this, and in fact it's plaguing our system.

**Review by KutnerUri — APPROVED** [review]

**Comment by KutnerUri** [comment]
I think this is good.

before:


after:
Screen Shot 2022-07-31 at 15 11 48

with headline before:
Screen Shot 2022-07-31 at 15 12 03

with headline after:
Screen Shot 2022-07-31 at 15 12 20

**Pull request merged**